test/test_prototype_builtin_datasets.py

```diff
@@ -10,6 +10,7 @@
 from torch.utils.data.graph import traverse
 from torchdata.datapipes.iter import Shuffler, ShardingFilter
 from torchvision.prototype import transforms, datasets
+from torchvision.prototype.datasets.utils._internal import TakerDataPipe
```

Comment:
@NivekT If I recall correctly, we wanted to upstream this to `torchdata`. Any progress on that?

Reply:
I will have a look at the implementation sometime today

Reply:
Please tag me in the PR so I can make the changes here after it is landed.